lib/CXGN/BrAPI/v2/ObservationVariables.pm

```diff
@@ -109,27 +109,53 @@ sub search {
             push @and_wheres, "db.db_id = '$_'";
         }
     }
-    if (scalar(@dbxref_ids)>0){
-        my @db_names;
-        my @dbxref_accessions;
-        foreach (@dbxref_ids){
-            my ($db_name, $accession) = split ':', $_;
-            push @db_names, $db_name;
-            push @dbxref_accessions, $accession;
-        }
-        foreach (@db_names){
-            push @and_wheres, "db.name = '$_'";
-        }
-        foreach (@dbxref_accessions){
-            push @and_wheres, "dbxref.accession = '$_'";
+
+    # External reference id and reference source search
+    if (scalar(@dbxref_ids) > 0 || scalar(@dbxref_terms)>0) {
+        my $cv_id = $self->trait_ontology_cv_id;
+
+        my @sub_and_wheres;
+        push @sub_and_wheres, "cvterm.cv_id = $cv_id";
+        push @sub_and_wheres, "reltype.name='VARIABLE_OF'";
+        if (scalar(@dbxref_ids)>0){
+            # TODO: Should this be OR?
```

Comment:
I ran into this with the programs and locations. There is an incompatibility between search and query parameters. The search endpoint parameters are OR whereas the query parameters are AND.